lib/cocoapods/generator/copy_resources_script.rb

```diff
@@ -89,7 +89,15 @@ def script
         end
 
         script += RSYNC_CALL
-        script += XCASSETS_COMPILE
+        script += XCASSETS_COMPILE_BEGIN
+        # TODO: Proper way of checking the env?
+        if ENV['XCASSETS_SELECTIVE_COMPILE'].is_set?
```

Comment:
I was suggesting to make this a test inside the bash script itself. Here it will only be checked when `pod install` runs versus the bash script in which it will run each time the script runs. The user can go add the "Build Setting" in the `.xcodeproj` and then re-run a clean build.